src/main/java/com/encora/Node.java

```diff
@@ -4,11 +4,11 @@
 import lombok.NonNull;
 
 @Data
-class Node {
+class Node<T> {
 
     @NonNull
-    private int value;
-    private Node left;
-    private Node right;
+    private T data;
```

Comment:
I don't disagree with changing `value` to `data`  but I think both names make sense.